pkg/linter/lintRuleSyntacticRedundantCommon.go

```diff
@@ -0,0 +1,229 @@
+	vpcResource  vpcmodel.VPC
+}
+
+////////////////////////////////////////////////////////////////////////////////////////////
+// functionality used by both SG and NACL lints
+////////////////////////////////////////////////////////////////////////////////////////////
+
+// Rule is syntactically redundant in SG if other rules in the table implies it
+// Rule is syntactically redundant in NACL if it is shadowed by higher priority rules of the table
+// A rule is shadowed in NACL if the rule will never determine the actual allow/deny status
+// (of any connection in any configuration)
+// A rule in a table has 3 dimensions: source, destination and connection
+// SG rule is redundant if the 3-dimensions union of the other rules contains it, thus rule is implied by others
+// NACL rule  is redundant if the 3-dimensions union of the higher priority (allow or deny) rules contains it
+// The above 3 dimensions containment is checked as follows:
+// in the below a "point" is a point in the src x dst x conn 3-dimensions
+// If each point in the rule is contained in the others/higher priority rules then it is redundant
+// This we check as follows:
+// For each table:
+//  1. partition the IPBlock to maximal atomic blocks w.r.t. the table
+//  2. For each rule r:
+//     2.1. Iterate over the cartesian product of r's src and dst atomic blocks
+//     For each item <src_atomic, dst_atomic>
+//     compute the union of the connection of the other/higher priority rules whose <src, dst> contains <src_atomic, dst_atomic>
+//     If the resulting conn does not contain the connection of the rule r then the block is not redundant
+//     Done? Rule is redundant
+//
+//nolint:gocyclo // better not split into two function
+func findRuleSyntacticRedundant(configs map[string]*vpcmodel.VPCConfig,
+	filterLayerName string) (res []ruleRedundant, err error) {
+	for _, config := range configs {
+		if config.IsMultipleVPCsConfig {
+			continue // no use in executing lint on dummy vpcs
+		}
+		filterLayer := config.GetFilterTrafficResourceOfKind(filterLayerName)
+		rules, err := filterLayer.GetRules()
+		if err != nil {
+			return nil, err
+		}
+		tableToRules, tableToAtomicBlocks := getTableOrientedStructs(rules)
+		// iterates over tables, in each table iterates over rules and finds those that are redundant (shadowed/implied)
+		for tableIndex, rules := range tableToRules {
+			tableAtomicBlocks := tableToAtomicBlocks[tableIndex]
+			for redundantRuleIndex := range rules {
+				// nacl - redundant if shadowed by over higher priority rules; sg - redundant if implied by "other" rule
+				// in the former case iterates only over relevant slice; in the latter skip the "myself" rule
+				var rulesToIterate []*vpcmodel.RuleOfFilter
+				if filterLayerName == vpcmodel.NaclLayer {
+					rulesToIterate = tableToRules[tableIndex][:redundantRuleIndex]
+				} else { // SG
+					rulesToIterate = tableToRules[tableIndex]
+				}
```

Comment:
rulesToIterate = append(tableToRules[tableIndex][:redundantRuleIndex], tableToRules[tableIndex][redundantRuleIndex+1:]...)

Reply:
not working, not sure it will make the code more readable. I'm looking at it, to be in the safe (understanding) side